cpp/src/arrow/flight/sql/client.cc

```diff
@@ -519,13 +519,17 @@ arrow::Result<std::shared_ptr<PreparedStatement>> PreparedStatement::ParseRespon
 
   std::shared_ptr<Schema> dataset_schema;
   if (!serialized_dataset_schema.empty()) {
-    io::BufferReader dataset_schema_reader(serialized_dataset_schema);
+    // Create a non-owned Buffer to avoid copying
```

Comment:
My above lifetime comment does not apply here since there's no way the deserialized `Schema` could reference memory in the buffer we're reading; zero copy lgtm